crates/matrix-sdk-store-encryption/src/lib.rs

```diff
@@ -54,6 +67,36 @@ pub enum Error {
     Length(usize, usize),
 }
 
+#[cfg(not(feature = "bincode"))]
+fn serialize<T: Serialize>(value: &T) -> Result<Vec<u8>, SerializationError> {
+    Ok(serde_json::to_vec(value)?)
+}
+
+#[cfg(feature = "bincode")]
+fn serialize<T: bincode::Encode>(value: &T) -> Result<Vec<u8>, SerializationError> {
+    Ok(bincode::encode_to_vec(value, bincode::config::standard())?)
+}
+
+#[cfg(not(feature = "bincode"))]
+fn deserialize<T>(value: &[u8]) -> Result<T, SerializationError>
+where
+    T: for<'de> Deserialize<'de>,
```

Comment:
```suggestion
    T: DeserializeOwned,
```

Reply:
The other instances of Deserialize trait bounds in the file also don’t use DeserializeOwned, but i can change that in another   commit

Reply:
I posted a PR for that earlier: #687

Reply:
I guess I duplicated the changes here then

Reply:
Yeah, but git should merge that just fine.